lib/util.js

```diff
@@ -9,39 +9,48 @@ const isError = internalUtil.isError;
 
 var Debug;
 
+function tryStringify(arg) {
+  try {
+    return JSON.stringify(arg);
+  } catch (_) {
+    return '[Circular]';
+  }
+}
+
 const formatRegExp = /%[sdj%]/g;
 exports.format = function(f) {
   if (typeof f !== 'string') {
-    var objects = [];
+    const objects = new Array(arguments.length);
     for (var index = 0; index < arguments.length; index++) {
-      objects.push(inspect(arguments[index]));
+      objects[index] = inspect(arguments[index]);
     }
     return objects.join(' ');
   }
 
   if (arguments.length === 1) return f;
 
-  var i = 1;
-  var args = arguments;
-  var len = args.length;
+  const len = arguments.length;
+  const args = new Array(len);
+  var i;
+  for (i = 0; i < len; i++) {
+    args[i] = arguments[i];
+  }
+
+  i = 1;
   var str = String(f).replace(formatRegExp, function(x) {
```

Comment:
`String(f)` here can probably just be replaced with just `f` since it should already be a string at that point.